src/components/dex/Trade.vue

```diff
@@ -357,14 +354,125 @@ export default {
       return this.side === 'Buy' ? this.percentForBuy(value) : this.percentForSell(value);
     },
 
-    percentForBuy() {
-      // TODO: fix this
-      console.log('return');
+    percentForBuy(value) {
+      if (!this.baseHolding || !this.baseHolding.availableBalance
+        || this.baseHolding.availableBalance.isLessThanOrEqualTo(0)) {
+        // TODO: Translate this
+        this.$services.alerts.error(`No balance of ${this.currentMarket.baseCurrency} available to Buy with`);
+        return '';
+      }
+
+      const baseAssetQuantity = this.baseHolding.availableBalance;
+
+      let newQuantity = new BigNumber(0);
+      const book = this.$store.state.orderBook.asks;
+      let orderPrice = null;
+      if (this.orderType !== 'Market' && this.$store.state.orderPrice !== '') {
+        orderPrice = new BigNumber(this.$store.state.orderPrice);
+      }
+
+      let willTakeOffers = false;
+      if (!orderPrice || orderPrice.isGreaterThanOrEqualTo(book[0].price)) {
+        willTakeOffers = !this.postOnly;
+      }
+
+      let leftToSpend = baseAssetQuantity;
+
+      if (willTakeOffers) {
+        book.forEach((level) => {
+          if (orderPrice && level.price.isGreaterThan(orderPrice)) {
+            return;
+          }
+
+          if (leftToSpend.isLessThanOrEqualTo(0)) {
+            return;
+          }
+
+          const levelCost = level.quantity.multipliedBy(level.price);
+
+          let spendAtThisLevel = levelCost.isGreaterThan(leftToSpend) ? leftToSpend : levelCost;
+          if (this.baseHolding.assetId === this.$store.state.currentNetwork.aph_hash) {
```

Comment:
Will this always be present?

Reply:
There is an earlier check for baseholding and I suppose if aph_hash doesn't exist then it shouldn't run that code in the if statement.